### PR TITLE
Makefile and CI flow clean up.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* [/bcgov/entity/5797](https://github.com/bcgov/entity/issues/5797)

*Description of changes:*

1. Add Makefile ;
2. CI flow using make commands;
3. Node version change from latest to 14.15.1;
4. Nginx version upgrade to 1.18.0;
5. Clean up Openshift templates and Jenkins;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Review by cameron-freshworks — APPROVED** [review]

File: Jenkinsfiles/build.groovy [hunk 1]
```diff
@@ -1,76 +1,99 @@
...
+//   See: https://docs.openshift.com/container-platform/3.9/using_images/other_images/jenkins.html for a complete list of JENKINS env vars
+
+import groovy.json.*
+
+// define constants - values sent in as env vars from whatever calls this pipeline
 def APP_NAME = 'namerequest-ui'
+def DESTINATION_TAG = 'dev'
+def TOOLS_TAG = 'tools'
+def NAMESPACE_BUILD = 'servicebc-ne-tools'
+def NAMESPACE_DEPLOY = '1rdehl-test'
```
Comment (severinbeauvais) [inline Jenkinsfiles/build.groovy:29]: Should this be dev?

File: Jenkinsfiles/build.groovy [hunk 2]
```diff
@@ -1,76 +1,99 @@
...
-// you'll need to change this to point to your application component's folder within your repository
-def CONTEXT_DIRECTORY = './client'
-
-// EDIT LINE BELOW (Add a reference to the CHAINED_BUILD_CONFIG)
-def CHAINED_BUILD_CONFIG = APP_NAME
-
-// The name of your deployment configuration; used to verify the deployment
-def DEPLOYMENT_CONFIG_NAME = APP_NAME
+// post a notification to rocketchat
+def rocketChatNotificaiton(token, channel, comments) {
```
Comment (severinbeauvais) [inline Jenkinsfiles/build.groovy:34]: Typo but I'm not sure if it matters.

File: Jenkinsfiles/build.groovy [hunk 3]
```diff
@@ -79,125 +102,105 @@ if( !triggerBuild(CONTEXT_DIRECTORY) ) {
...
-            def IMAGE_HASH = getImageTagHash("${IMAGESTREAM_NAME}")
-            echo "IMAGE_HASH: ${IMAGE_HASH}"
-            openshift.tag("${IMAGESTREAM_NAME}@${IMAGE_HASH}", "${IMAGESTREAM_NAME}:${DEPLOY_TAG}")
-          }
+            ROCKETCHAT_TOKEN = sh (
+                    script: """oc get secret/apitest-secrets -n ${NAMESPACE_BUILD} -o template --template="{{.data.ROCKETCHAT_TOKEN}}" | base64 --decode""",
+                        returnStdout: true).trim()
 
-          echo "Deployment Complete."
+            rocketChatNotificaiton("${ROCKETCHAT_TOKEN}", "${ROCKETCHAT_DEVELOPER_CHANNEL}", "${APP_NAME} build and deploy to ${DESTINATION_TAG} ${currentBuild.result}!")
```
Comment (severinbeauvais) [inline Jenkinsfiles/build.groovy:202]: Ah, the matching typo :)

File: Jenkinsfiles/deploy-prod.groovy [hunk 4]
```diff
@@ -1,85 +1,169 @@
...
 def SOURCE_TAG = 'test'
-def DEPLOY_TAG = 'prod'
-def PREV_TAG = "${DEPLOY_TAG}-previous"
+def DESTINATION_TAG = 'prod'
+def TOOLS_TAG = 'tools'
 
-// define groovy functions
-import groovy.json.JsonOutput
+def NAMESPACE_BUILD = 'servicebc-ne-tools'
+def NAMESPACE_DEPLOY = '1rdehl-test'
```
Comment (severinbeauvais) [inline Jenkinsfiles/deploy-prod.groovy:31]: Should this be prod?

File: Jenkinsfiles/build.groovy (hunk 1 above)
Comment (pwei1018) [reply Jenkinsfiles/build.groovy]: Fixed.

File: Jenkinsfiles/build.groovy (hunk 2 above)
Comment (pwei1018) [reply Jenkinsfiles/build.groovy]: Fixed.

File: Jenkinsfiles/build.groovy (hunk 3 above)
Comment (pwei1018) [reply Jenkinsfiles/build.groovy]: Fixed.

File: Jenkinsfiles/deploy-prod.groovy (hunk 4 above)
Comment (pwei1018) [reply Jenkinsfiles/deploy-prod.groovy]: Fixed

**Review by severinbeauvais — APPROVED** [review]

**Comment by cameron-freshworks** [comment]
@pwei1018 Did you want to merge this before our next promotion to TEST env? 
2-230pm this afternoon.

**Pull request merged**